std/textproto/reader.ts

```diff
@@ -0,0 +1,227 @@
+// Copyright 2018-2020 the Deno authors. All rights reserved. MIT license.
+// Based on https://github.com/golang/go/tree/master/src/net/textproto
+// Copyright 2009 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+import type { BufReader } from "../io/bufio.ts";
+import { concat } from "../bytes/mod.ts";
+import { decode } from "../encoding/utf8.ts";
+
+// FROM https://github.com/denoland/deno/blob/b34628a26ab0187a827aa4ebe256e23178e25d39/cli/js/web/headers.ts#L9
+const invalidHeaderCharRegex = /[^\t\x20-\x7e\x80-\xff]/g;
+
+function str(buf: Uint8Array | null | undefined): string {
+  if (buf == null) {
+    return "";
+  } else {
+    return decode(buf);
+  }
+}
+
+function charCode(s: string): number {
+  return s.charCodeAt(0);
+}
+
+type CodeLine = [
+  number, // code
+  boolean, // continued
+  string, // message
+];
+export type Response = [number, string];
+
+export class TextProtoReader {
+  constructor(readonly r: BufReader) {}
+
+  /** readLine() reads a single line from the TextProtoReader,
+   * eliding the final \n or \r\n from the returned string.
+   */
+  async readLine(): Promise<string | null> {
+    const s = await this.readLineSlice();
+    if (s === null) return null;
```

Comment:
Should this have curly braces around to be more consistant with the style in the rest of the file?